### PR TITLE
Add Eslint to project

### PR DESCRIPTION
**Pull request by JuliaKirschenheuter**
Add @nextcloud/eslint-confg
Add logger
Add https://github.com/nextcloud/.github/blob/master/workflow-templates/lint-eslint.yml
Add https://github.com/nextcloud/.github/blob/master/workflow-templates/lint-eslint-when-unrelated.yml

## TODO

- [ ] Make linter check required

### CODE REVIEW

**Comment by JuliaKirschenheuter** [comment]
@skjnldsv i would like to ask you about this configuration. ESLint is working, but if i run `npm run lint:fix` i have 9 errors:
`'logger' is not defined  no-undef`
and 8 of them `'ICAL' is not defined `

please let me know if  you have some idea, how could i fix it.
Thank you!

**Comment by ChristophWurst** [comment]
The missing logger can be fixed by adding https://github.com/nextcloud/nextcloud-logger/ to this project similar to https://github.com/nextcloud/mail/blob/main/src/logger.js

The ICAL error is weird. We use `ical.js` for the import.

File: src/utils/logger.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { getLoggerBuilder } from '@nextcloud/logger'
+
+export default getLoggerBuilder()
+	.detectUser()
+	.setApp('@nextcloud/vue')
```
Comment (ChristophWurst) [inline src/utils/logger.js:26]:
```suggestion
	.setApp('@nextcloud/calendar-availability-vue')
```

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+module.exports = {
+	globals: {
+		EMOJIS: true,
+		SCOPE_VERSION: true,
+		TRANSLATIONS: true,
+		oc_userconfig: true,
+		appName: true,
+		appVersion: true,
```
Comment (ChristophWurst) [inline .eslintrc.js:8]: we don't use any of this, do we?

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@master
+      - name: Set up php${{ matrix.php-versions }}
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: ${{ matrix.php-versions }}
+          coverage: none
+      - name: Lint
+        run: composer run lint
```
Comment (ChristophWurst) [inline .github/workflows/lint.yml:33]: there are no php files

File: .github/workflows/lint.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        uses: actions/checkout@master
+      - name: Set up php
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: 7.4
+          coverage: none
+      - name: Install dependencies
+        run: composer i
+      - name: Run coding standards check
+        run: composer run cs:check
```
Comment (ChristophWurst) [inline .github/workflows/lint.yml:49]: only relevant for php

**Review by ChristophWurst — CHANGES_REQUESTED** [review]

**Comment by skjnldsv** [comment]
@JuliaKirschenheuter please use the lin templates from https://github.com/nextcloud/.github

**Comment by skjnldsv** [comment]
> The ICAL error is weird. We use `ical.js` for the import at

I was about to write 
> iirc the ICAL pollute window and uses it.
> Make sure it's defiend as a global 

But I realised contacts works fine without :thinking: https://github.com/nextcloud/contacts/blob/9d1a7cf81a5420ddb9d6ea8961931d046b3e6e2b/.eslintrc.js#L2-L5
So I guess it's something else

File: stylelint.config.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+const stylelintConfig = require('@nextcloud/stylelint-config')
```
[the remark this answers is not in the document]
Comment (st3iny) [reply stylelint.config.js]: Done

File: package.json [hunk 6]
```diff
@@ -29,24 +33,33 @@
...
     "eslint": "^8.2.0",
     "postcss": "^8.3.11",
     "rollup": "^2.60.0",
     "rollup-plugin-peer-deps-external": "^2.2.4",
+    "css-loader": "^5.2.7",
+    "file-loader": "^6.2.0",
+    "prettier": "^2.5.1",
     "rollup-plugin-postcss": "^4.0.1",
     "rollup-plugin-vue": "^5.1.9",
+    "vue-loader": "^15.9.8",
```
[the remark this answers is not in the document]
Comment (st3iny) [reply package.json]: Done

File: package.json [hunk 7]
```diff
@@ -7,7 +7,11 @@
   "scripts": {
     "build": "rollup -c",
     "watch": "rollup -c -w",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "lint": "eslint --ext .js,.vue src",
+    "lint:fix": "eslint --ext .js,.vue src --fix",
+    "stylelint": "stylelint src/**/*.vue src/**/*.scss src/**/*.css",
+    "stylelint:fix": "stylelint src/**/*.vue src/**/*.scss src/**/*.css --fix"
```
[the remark this answers is not in the document]
Comment (st3iny) [reply package.json]: Done

File: .stylelintignore [hunk 8]
```diff
@@ -0,0 +1 @@
+/src/fonts
```
[the remark this answers is not in the document]
Comment (st3iny) [reply .stylelintignore]: Done

File: .github/workflows/lint.yml (hunk 4 above)
Comment (st3iny) [reply .github/workflows/lint.yml]: Done

File: .github/workflows/lint.yml [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@master
+      - name: Download schema
+        run: wget https://apps.nextcloud.com/schema/apps/info.xsd
+      - name: Lint info.xml
+        uses: ChristophWurst/xmllint-action@v1
+        with:
+          xml-file: ./appinfo/info.xml
+          xml-schema-file: ./info.xsd
```
[the remark this answers is not in the document]
Comment (st3iny) [reply .github/workflows/lint.yml]: Done

**Review by ChristophWurst — APPROVED** [review]
:shipit:

**Pull request merged**